### PR TITLE
Big update of the library

### PR DESCRIPTION
**Pull request by achedeuzot**
This PR adds quite a big update of all the dependencies as well as adds a configurable JSON parser (following the logic from Ueberauth). 

The paths are on a separate repo as I don't know if this will be merged or not for now. If the merge is planned, I'll update the paths back to `sntran`.

### CODE REVIEW

File: .formatter.exs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[
+  inputs: ["*.{ex,exs}", "{config,lib,test}/**/*.{ex,exs}"]
```
Comment (hez) [inline .formatter.exs]: 👍

File: README.md [hunk 2]
```diff
@@ -15,7 +21,7 @@
 
   ```elixir
   def deps do
-    [{:ueberauth_auth0, "~> 0.3"}]
+    [{:ueberauth_auth0, github: "vivakit/ueberauth_auth0", branch: "big-update"}]
```
Comment (hez) [inline README.md]: Kinda hard to merge this in with all the changes to the github urls. 😭

File: lib/ueberauth/strategy/auth0.ex [hunk 3]
```diff
@@ -16,16 +16,17 @@ defmodule Ueberauth.Strategy.Auth0 do
         providers: [
           auth0: { Ueberauth.Strategy.Auth0, [default_scope: "openid profile email"] }
         ]
-  Deafult is `"openid profile email"`
+  Default is `"openid profile email"`
```
Comment (hez) [inline lib/ueberauth/strategy/auth0.ex]: 👍

**Comment by hez** [comment]
This PR became quite large because of the inclusion of the formatter config, I'd suggest making that a separate PR for clarity sake. I'd love to see this PR get accepted, let me know if there is anything we can do to help.

@sntran what are you thoughts on this?

File: README.md (hunk 2 above)
Comment (sntran) [reply README.md]: Yeah. If this PR is for this repo, then it does not make sense to change the Github URL.

**Comment by sntran** [comment]
I would prefer separate PR for separate things. This PR includes 3 different things (updating deps, option for JSON parser, formatter and dialyzer).

As I also commented, if you want to merge the code, then the GitHub URLs need to stay the same.

Also, while I appreciate your efforts handling all these pending changes, I would like to wait until @achedeuzot becomes an actual maintainer before adding their name to `mix.exs`. That can definitely happen when @achedeuzot can resolve the issues here.

**Comment by achedeuzot** [comment]
Ok, so I separated this into two PRs: #75 and #76. I'll close this one.

**Comment by hez** [comment]
@achedeuzot that was really quick turn around, awesome stuff 👍

**Pull request closed without merging**